client/app/pages/destinations/DestinationsList.jsx

```diff
@@ -57,16 +57,11 @@ class DestinationsList extends React.Component {
     const target = { options: {}, type: selectedType.type };
     helper.updateTargetWithValues(target, values);
 
-    return Destination.create(target)
-      .then(destination => {
-        this.setState({ loading: true });
-        Destination.query().then(destinations => this.setState({ destinations, loading: false }));
-        return destination;
-      })
-      .catch(error => {
-        const message = find([get(error, "response.data.message"), get(error, "message"), "Failed saving."], isString);
-        return Promise.reject(new Error(message));
-      });
```

Comment:
`getErrorMessage` already handles this

Reply:
But it wont fallback to ""Failed saving.", right? `getErrorMessage` would return `It seems like we encountered an error. Try refreshing this page or contact your administrator.`. Maybe, let's add an optional parameter to `getErrorMessage` for default message which will default to existing string, and can be redefined when needed, WDYT?

Reply:
> But it wont fallback to Failed saving., right?

That didn't seem to be much harm 😅, also it would be in accordance with the Data Source one.

Makes sense to make this message replaceable anyway

Reply:
Up to you. I just noticed this, but for me it doesn't feel _that_ important. Rather it could be nice to show "friendlier" error message to user.

Reply:
https://github.com/getredash/redash/pull/4959/commits/828e468a89331673268bee0484a60e8281bd4e76 🙂